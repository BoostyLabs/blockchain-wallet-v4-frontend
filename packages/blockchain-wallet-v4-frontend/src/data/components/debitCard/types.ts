// State
import { RemoteDataType } from '@core/remote/types'

export type DebitCardState = {
  cardCreationData: RemoteDataType<string, string>
  cardToken: string
  cards: RemoteDataType<string, Array<DebitCardType>>
  currentCardAccount: RemoteDataType<string, AccountType>
  currentCardSelected: DebitCardType | undefined
  eligibleAccounts: RemoteDataType<string, Array<AccountType>>
  lockHandler: RemoteDataType<string, boolean>
  products: Array<ProductType>
  selectAccountHandler: RemoteDataType<string, string>
  terminateHandler: RemoteDataType<string, string>
}

export type ProductType = {
  brand: string
  price: {
    symbol: string
    value: number
  }
  productCode: string
  type: string
}

export type DebitCardType = {
  brand: string
  createdAt: string
  expiry: string
  id: string
  last4: string
  status: CardStateType
  type: string
}

export type CardActionType = {
  id: string
  newLockState: boolean
}

export enum CardStateType {
  ACTIVE = 'ACTIVE',
  LOCKED = 'LOCKED',
  TERMINATED = 'TERMINATED'
}

export type BalanceType = {
  symbol: string
  value: string
}

export type AccountType = {
  balance: BalanceType
}
