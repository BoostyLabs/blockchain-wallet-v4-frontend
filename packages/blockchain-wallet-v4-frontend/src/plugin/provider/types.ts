import { providers } from 'ethers'

export interface RequestArguments {
  /** The RPC method to request. */
  method: string

  /** The params of the RPC method, if any. */
  params?: unknown[] | Record<string, unknown> | providers.TransactionRequest
}

export interface ProviderMessage {
  readonly data: unknown
  readonly type: string
}

export type ProviderMessageData = string & {
  code: number
  message: string
}

export interface ProviderConnectInfo {
  readonly chainId: string
}

export type Maybe<T> = Partial<T> | null | undefined

export enum ConnectionEvents {
  Connected = 'connected',
  Disconnected = 'disconnected',
  Error = 'error'
}

export enum StandardEvents {
  AccountsChanged = 'accountsChanged',
  ChainChanged = 'chainChanged'
}
