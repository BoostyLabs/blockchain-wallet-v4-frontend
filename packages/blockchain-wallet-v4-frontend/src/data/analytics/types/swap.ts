export enum Events {
  SWAP_EARN_REWARDS_BUTTON_CLICKED = 'Swap Earn Rewards Clicked',
  SWAP_EARN_REWARDS_BUTTON_VIEWED = 'Swap Earn Rewards Viewed'
}

type SwapEarnRewardsButtonClick = {
  key: Events.SWAP_EARN_REWARDS_BUTTON_CLICKED
  properties: {
    currency: string
    device: string
  }
}

type SwapEarnRewardsButtonView = {
  key: Events.SWAP_EARN_REWARDS_BUTTON_VIEWED
  properties: {
    currency: string
    device: string
  }
}

export type TrackEventAction = SwapEarnRewardsButtonClick | SwapEarnRewardsButtonView
