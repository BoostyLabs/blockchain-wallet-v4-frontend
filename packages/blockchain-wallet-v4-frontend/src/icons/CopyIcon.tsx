import React from 'react'

const CopyIcon = ({ color = '#000', size = 24 }: { color?: string; size?: number }) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox='0 0 14 14'
      fill={color}
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M11.9335 3.86634H10.1335V2.06634C10.1335 1.13301 9.3335 0.333008 8.40016 0.333008H2.06683C1.06683 0.333008 0.333496 1.13301 0.333496 2.06634V8.39967C0.333496 9.33301 1.1335 10.133 2.06683 10.133H3.86683V11.933C3.86683 12.8663 4.66683 13.6663 5.60016 13.6663H11.9335C12.8668 13.6663 13.6668 12.8663 13.6668 11.933V5.59967C13.6668 4.66634 12.8668 3.86634 11.9335 3.86634ZM2.06683 8.79967C1.86683 8.79967 1.66683 8.59967 1.66683 8.39967V2.06634C1.66683 1.86634 1.80016 1.66634 2.06683 1.66634H8.40016C8.60016 1.66634 8.80016 1.86634 8.80016 2.06634V3.86634H5.60016C4.66683 3.86634 3.86683 4.66634 3.86683 5.59967V8.79967H2.06683ZM12.3335 11.933C12.3335 12.133 12.1335 12.333 11.9335 12.333H5.60016C5.40016 12.333 5.20016 12.133 5.20016 11.933V10.133V5.59967C5.20016 5.39967 5.40016 5.19967 5.60016 5.19967H10.1335H11.9335C12.1335 5.19967 12.3335 5.39967 12.3335 5.59967V11.933Z'
        fill='#98A1B2'
      />
    </svg>
  )
}

export default CopyIcon
