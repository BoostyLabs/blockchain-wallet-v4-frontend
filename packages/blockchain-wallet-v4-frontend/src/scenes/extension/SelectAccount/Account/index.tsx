import React, { MouseEvent } from 'react'
import { IconCheckCircle, IconCopy } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

import IconBitcoin from '../../../../icons/IconBitcoin'
import IconEthereum from '../../../../icons/IconEthereum'
import { SwapAccountType } from '..'

export const AccountBlock = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px 0;
  cursor: pointer;
`
const AccountInfo = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto auto auto 16px;
`
const WalletBlock = styled.div`
  display: flex;
  align-items: center;
  cursor: copy;
`
type AccountProps = {
  account: SwapAccountType
  activeAccountIndex: number
  setActiveAccountIndex: (activeAccountIndex: number) => void
}

const Account: React.FC<AccountProps> = ({
  account,
  activeAccountIndex,
  setActiveAccountIndex
}) => {
  const copyAddress = (event: MouseEvent<HTMLDivElement>, address: string | number): void => {
    event.stopPropagation()
    navigator.clipboard.writeText(address.toString())
    // TODO: Some notification about copy address.
  }

  const shortingAddress = (address: string | number): string => {
    const addressString: string = address.toString()
    return `${addressString.slice(0, 4)}...${addressString.slice(-4)}`
  }

  return (
    <AccountBlock
      key={account.accountIndex}
      onClick={() => setActiveAccountIndex(account.accountIndex)}
      className='account'
    >
      {account.coin === 'ETH' ? <IconEthereum /> : <IconBitcoin />}
      <AccountInfo>
        <Text size='12px' weight={500} style={{ color: '#98A1B2', lineHeight: '150%' }}>
          {account.label}
        </Text>
        <WalletBlock
          onClick={(event: MouseEvent<HTMLDivElement>) => copyAddress(event, account.address)}
          className='wallet-address'
        >
          <Text size='14px' color='white' weight={500} style={{ lineHeight: '150%' }}>
            {shortingAddress(account.address)}
          </Text>
          <IconCopy color='#ffffff' height='12px' width='12px' style={{ paddingLeft: '5px' }} />
        </WalletBlock>
        <Text size='14px' color='white' weight={500} style={{ lineHeight: '150%' }}>
          {account.balance} {account.coin}
        </Text>
        <Text size='12px' weight={500} style={{ color: '#98A1B2', lineHeight: '150%' }}>
          $5,225.01
        </Text>
      </AccountInfo>
      {activeAccountIndex === account.accountIndex && (
        <IconCheckCircle height='24px' width='24px' style={{ fill: '#65A5FF' }} />
      )}
    </AccountBlock>
  )
}

export default Account
