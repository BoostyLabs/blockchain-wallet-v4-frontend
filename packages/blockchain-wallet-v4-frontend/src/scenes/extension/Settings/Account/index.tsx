import React from 'react'
import { Icon } from '@blockchain-com/constellation'
import { IconChevronRightV2 } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Setting, SettingsHeading } from '..'
import { SettingsLink, SettingsLinkLabel, SettingsList } from '../Overal'

const AccountSettingLabel = styled(SettingsLinkLabel)`
  margin-left: 0;
`

export const Account = (props) => {
  const { path } = props.match

  const accountSettings = [
    new Setting('Import account', `${path}/import`),
    new Setting('Upgrade account', `${path}/upgrade`)
  ]
  return (
    <>
      <SettingsHeading>Account</SettingsHeading>
      <SettingsList>
        {accountSettings.map((setting: Setting) => (
          <li key={setting.label}>
            <SettingsLink to={setting.path}>
              <AccountSettingLabel>{setting.label}</AccountSettingLabel>
              <Icon color='white800' label='IconBack' size='md'>
                <IconChevronRightV2 />
              </Icon>
            </SettingsLink>
          </li>
        ))}
      </SettingsList>
    </>
  )
}
