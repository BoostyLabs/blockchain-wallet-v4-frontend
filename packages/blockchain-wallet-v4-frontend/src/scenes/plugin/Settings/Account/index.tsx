import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Icon } from '@blockchain-com/constellation'
import { IconChevronRightV2 } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Flex } from 'components/Flex'

import { Setting, SettingsHeading } from '..'
import { SettingsLink, SettingsLinkLabel, SettingsList } from '../Overal'

const AccountSettingLabel = styled(SettingsLinkLabel)`
  margin-left: 0;
`

export const Account = (props) => {
  const { path } = props.match

  const accountSettings = [
    new Setting('', 'Import account', `${path}/import`),
    new Setting('', 'Upgrade account', `${path}/upgrade`)
  ]
  return (
    <>
      <SettingsHeading>
        <FormattedMessage id='scenes.plugin.settings.account.heading' defaultMessage='Account' />
      </SettingsHeading>
      <SettingsList>
        {accountSettings.map((setting: Setting) => (
          <li key={setting.label}>
            <SettingsLink to={setting.path}>
              <Flex justifyContent='space-between' alignItems='center'>
                <AccountSettingLabel>{setting.label}</AccountSettingLabel>
                <Icon color='white800' label='IconBack' size='md'>
                  <IconChevronRightV2 />
                </Icon>
              </Flex>
            </SettingsLink>
          </li>
        ))}
      </SettingsList>
    </>
  )
}
