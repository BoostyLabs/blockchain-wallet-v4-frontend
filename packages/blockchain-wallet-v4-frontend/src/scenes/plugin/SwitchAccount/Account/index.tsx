import React, { MouseEvent, useEffect, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import { connect, useDispatch, useSelector } from 'react-redux'
import { IconCheckCircle, IconCopy } from '@blockchain-com/icons'
import { SwapAccountType } from 'blockchain-wallet-v4-frontend/src/data/components/swap/types'
import Tooltip from 'blockchain-wallet-v4-frontend/src/scenes/plugin/SwitchAccount/Tooltip'
import { bindActionCreators } from 'redux'
import styled from 'styled-components'

import { Icon, Text } from 'blockchain-info-components'
import CoinDisplay from 'components/Display/CoinDisplay'
import FiatDisplay from 'components/Display/FiatDisplay'
import { Flex } from 'components/Flex'
import { actions, selectors } from 'data'
import { actions as cacheActions } from 'data/cache/slice'
import { RootState } from 'data/rootReducer'

const AccountBlock = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 22px 0;
  cursor: pointer;
`
const ConnectBlock = styled.div`
  position: relative;
  #tooltip {
    display: none;
  }
  &:hover {
    #tooltip {
      display: block;
    }
  }
`
const AccountInfo = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  margin: auto auto auto 16px;
  #tooltip {
    display: none;
  }
  &:hover {
    #tooltip {
      display: block;
    }
  }
`
const IconCheckWrapper = styled(IconCheckCircle)`
  path {
    fill: ${(props) => props.theme.white};
  }
`
const IconUncheckCircle = styled.div`
  width: 24px;
  height: 24px;
  border-radius: 50%;
  border: 2px solid ${(props) => props.theme.grey800};
`
const WalletBlock = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  cursor: copy;
`
const CoinTextBlock = styled(Text)`
  display: flex;
`

interface TooltipProperties {
  backgroundColor: string
  index: number
  leftBlockPosition: number
  leftTrianglePosition: number
  text: string
  textColor: string
}

type AccountProps = {
  account: SwapAccountType[]
  copiedAccountIndex: number
  index: number
  selectedAccountIndex: number
  setCopiedAccountIndex: (copiedAccountIndex: number) => void
  setSelectedAccountIndex: (selectedAccountIndex: number) => void
}

const defaultTooltipProperties: TooltipProperties = {
  backgroundColor: 'black',
  index: 0,
  leftBlockPosition: 50,
  leftTrianglePosition: 50,
  text: 'Copy to clipboard',
  textColor: 'grey400'
}

const Account: React.FC<AccountProps & any> = ({
  account,
  addressR,
  copiedAccountIndex,
  index,
  setCopiedAccountIndex,
  setIsSwitchAccountVisible,
  ...props
}) => {
  const dispatch = useDispatch()
  const selectedAccount = useSelector((state) => selectors.cache.getCache(state).selectedAccount)
  const state = useSelector((state: RootState) => state)

  const balance = selectors.balances.getCoinTotalBalance(account[0].coin)(state).getOrElse(0)

  const [tooltipProperties, setTooltipProperties] =
    useState<TooltipProperties>(defaultTooltipProperties)

  useEffect(() => {
    props.requestActions.getNextAddress(account[0])
  }, [])

  const changeTooltipProperties = (
    backgroundColor: string,
    index: number,
    leftBlockPosition: number,
    leftTrianglePosition: number,
    text: string,
    textColor: string
  ): TooltipProperties => {
    return { backgroundColor, index, leftBlockPosition, leftTrianglePosition, text, textColor }
  }

  const generateNextAddress = () => {
    return addressR.cata({
      Failure: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Please try again', 'black')
        ),
      Loading: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Getting address...', 'black')
        ),
      NotAsked: () =>
        setTooltipProperties(
          changeTooltipProperties('white', index, 50, 50, 'Please try again', 'black')
        ),
      Success: (val) => val.address
    })
  }

  const copyAddress = (event: MouseEvent<HTMLDivElement>) => {
    event.stopPropagation()
    setCopiedAccountIndex(index)
    const accountAddress = generateNextAddress()

    if (accountAddress) {
      navigator.clipboard.writeText(accountAddress)
      setTooltipProperties(changeTooltipProperties('white', index, 50, 50, 'Copied!', 'black'))
    }
  }

  const selectAccount = (account) => {
    dispatch(cacheActions.setSelectedAccount(account))
    setIsSwitchAccountVisible(false)
  }

  return (
    <AccountBlock key={index} onClick={() => selectAccount(account)}>
      <Icon size='24' name={account[0].coin} />
      <AccountInfo>
        {copiedAccountIndex === index ? (
          <Tooltip tooltipProperties={tooltipProperties} />
        ) : (
          <Tooltip tooltipProperties={defaultTooltipProperties} />
        )}
        <WalletBlock onClick={(event: MouseEvent<HTMLDivElement>) => copyAddress(event)}>
          <Text
            size='16px'
            color='white'
            weight={600}
            lineHeight='150%'
            style={{ marginRight: '9px' }}
          >
            {account.label ? (
              <FormattedMessage
                id={`plugin.switch.account.label.${account[0].label}`}
                defaultMessage={account[0].label}
              />
            ) : (
              <FormattedMessage
                id='plugin.switch.account.private_key_wallet'
                defaultMessage='Private Key Wallet'
              />
            )}
          </Text>
          <IconCopy width={16} height={16} />
        </WalletBlock>
        <Flex>
          <FiatDisplay color='grey400' size='12px' weight={500} coin={account[0].coin}>
            {balance}
          </FiatDisplay>
          <CoinTextBlock color='grey400' size='12px' weight={500}>
            &nbsp;(
            <CoinDisplay coin={account[0].coin} color='grey400' size='12px' weight={500}>
              {balance}
            </CoinDisplay>
            )
          </CoinTextBlock>
        </Flex>
      </AccountInfo>
      {selectedAccount && selectedAccount[0].coin === account[0].coin ? (
        <ConnectBlock>
          <IconCheckWrapper height='24px' width='24px' />
          <Tooltip
            tooltipProperties={changeTooltipProperties(
              'white',
              index,
              70,
              65,
              'Connected!',
              'black'
            )}
          />
        </ConnectBlock>
      ) : (
        <ConnectBlock>
          <IconUncheckCircle />
          <Tooltip
            tooltipProperties={changeTooltipProperties(
              'black',
              index,
              50,
              50,
              'Connect',
              'grey400'
            )}
          />
        </ConnectBlock>
      )}
    </AccountBlock>
  )
}

const mapStateToProps = (state, props) => ({
  addressR: selectors.components.request.getNextAddress(state, props.account[0])
})

const mapDispatchToProps = (dispatch) => ({
  requestActions: bindActionCreators(actions.components.request, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(Account)
