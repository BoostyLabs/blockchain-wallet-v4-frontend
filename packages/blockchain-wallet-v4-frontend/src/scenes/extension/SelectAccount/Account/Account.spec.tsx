import React from 'react'
import { IconCheckCircle } from '@blockchain-com/icons'
import { shallow } from 'enzyme'

import { SwapBaseCounterTypes } from 'data/components/swap/types'

import Account, { AccountBlock } from '.'

describe('Account', () => {
  const account = {
    accountIndex: 0,
    address: '0xb0106c26a6CfbAFB372e317f1a535304F69D3968',
    archived: false,
    balance: '5.02',
    baseCoin: 'Ethereum',
    coin: 'ETH',
    index: 0,
    label: 'Ethereum account',
    type: SwapBaseCounterTypes.CUSTODIAL
  }
  const activeAccountIndex = 1
  const setActiveAccountIndex = jest.fn()

  const accountComponent = shallow(
    <Account
      account={account}
      activeAccountIndex={activeAccountIndex}
      setActiveAccountIndex={setActiveAccountIndex}
    />
  )
  it('Should display the one account block', () => {
    expect(accountComponent.find(AccountBlock)).toHaveLength(1)
  })

  it('Should display the copy address button', () => {
    const onCopyAddressButton = accountComponent.find('.wallet-address')

    expect(onCopyAddressButton).toHaveLength(1)
  })

  it('Should copy wallet address after click', () => {
    const onCopyAddressButton = accountComponent.find('.wallet-address')

    Object.assign(window.navigator, {
      clipboard: {
        writeText: jest.fn().mockImplementation(() => Promise.resolve())
      }
    })

    onCopyAddressButton.simulate('click', {
      stopPropagation: () => {},
      writeText: window.navigator.clipboard.writeText('address')
    })

    expect(navigator.clipboard.writeText).toHaveBeenCalled()
    expect(window.navigator.clipboard.writeText).toHaveBeenCalledWith('address')
    expect(accountComponent).toMatchSnapshot()
  })

  it('Should display the icon of the selected account', () => {
    accountComponent.find('.account').simulate('click')
    expect(accountComponent.find(IconCheckCircle)).toBeTruthy()
  })
})
