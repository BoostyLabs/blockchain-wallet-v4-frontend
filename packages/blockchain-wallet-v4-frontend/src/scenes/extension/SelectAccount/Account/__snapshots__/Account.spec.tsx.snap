// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Should copy wallet address after click 1`] = `
<styled.div
  className="account"
  key="0"
  onClick={[Function]}
>
  <IconEthereum />
  <styled.div>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="12px"
      style={
        Object {
          "color": "#98A1B2",
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      Ethereum account
    </Text>
    <styled.div
      className="wallet-address"
      onClick={[Function]}
    >
      <Text
        altFont={false}
        capitalize={false}
        color="white"
        cursor="inherit"
        italic={false}
        lineHeight="inherit"
        opacity={1}
        size="14px"
        style={
          Object {
            "lineHeight": "150%",
          }
        }
        uppercase={false}
        weight={500}
      >
        0xb0...3968
      </Text>
      <SvgIconCopy
        color="#ffffff"
        height="12px"
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
        width="12px"
      />
    </styled.div>
    <Text
      altFont={false}
      capitalize={false}
      color="white"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="14px"
      style={
        Object {
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      5.02
       
      ETH
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="12px"
      style={
        Object {
          "color": "#98A1B2",
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      $5,225.01
    </Text>
  </styled.div>
</styled.div>
`;

exports[`Account renders correctly 1`] = `
<styled.div
  className="account"
  key="0"
  onClick={[Function]}
>
  <IconEthereum />
  <styled.div>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="12px"
      style={
        Object {
          "color": "#98A1B2",
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      Ethereum account
    </Text>
    <styled.div
      className="wallet-address"
      onClick={[Function]}
    >
      <Text
        altFont={false}
        capitalize={false}
        color="white"
        cursor="inherit"
        italic={false}
        lineHeight="inherit"
        opacity={1}
        size="14px"
        style={
          Object {
            "lineHeight": "150%",
          }
        }
        uppercase={false}
        weight={500}
      >
        0xb0...3968
      </Text>
      <SvgIconCopy
        color="#ffffff"
        height="12px"
        style={
          Object {
            "paddingLeft": "5px",
          }
        }
        width="12px"
      />
    </styled.div>
    <Text
      altFont={false}
      capitalize={false}
      color="white"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="14px"
      style={
        Object {
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      5.02
       
      ETH
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="inherit"
      opacity={1}
      size="12px"
      style={
        Object {
          "color": "#98A1B2",
          "lineHeight": "150%",
        }
      }
      uppercase={false}
      weight={500}
    >
      $5,225.01
    </Text>
  </styled.div>
</styled.div>
`;
